src/variety/kuroclone.js

```diff
@@ -0,0 +1,232 @@
+//
+// パズル固有スクリプト部 クロクローン版 kuroclone.js
+//
+
+/* global Set:false */
+
+(function(pidlist, classbase) {
+	if (typeof module === "object" && module.exports) {
+		module.exports = [pidlist, classbase];
+	} else {
+		pzpr.classmgr.makeCustom(pidlist, classbase);
+	}
+})(["kuroclone"], {
+	//---------------------------------------------------------
+	// マウス入力系
+	MouseEvent: {
+		use: true,
+		inputModes: {
+			edit: ["border", "number", "direc", "clear"],
+			play: ["shade", "unshade"]
+		},
+		mouseinput_auto: function() {
+			if (this.puzzle.playmode) {
+				if (this.mousestart || this.mousemove) {
+					this.inputcell();
+				}
+			} else if (this.puzzle.editmode) {
+				if (this.mousestart || this.mousemove) {
+					if (this.isBorderMode()) {
+						this.inputborder();
+					} else {
+						this.inputdirec();
+					}
+				} else if (this.mouseend && this.notInputted()) {
+					this.inputqnum();
+				}
+			}
+		}
+	},
+
+	//---------------------------------------------------------
+	// キーボード入力系
+	KeyEvent: {
+		enablemake: true,
+		moveTarget: function(ca) {
+			if (ca.match(/shift/)) {
+				return false;
+			}
+			return this.moveTCell(ca);
+		},
+
+		keyinput: function(ca) {
+			if (this.key_inputdirec(ca)) {
+				return;
+			}
+			this.key_inputqnum(ca);
+		}
+	},
+
+	//---------------------------------------------------------
+	// 盤面管理系
+	Cell: {
+		numberRemainsUnshaded: true,
+		minmum: 1,
+		maxnum: 99
```

Comment:
Could you change this to a function that uses the room size?

Reply:
I've tried, but it seems difficult...
In kuroclone, a maxnum depends on the room size of next cell (**possibly another room**), not the current room size.
So, qdir must be decided before qnum, but it cannot.

Reply:
Good point. Could you change it to a function that uses half the board size instead?